src/test/kotlin/org/rust/ide/inspections/lints/RsUnknownCrateTypesInspectionTest.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import org.rust.ide.inspections.RsInspectionsTestBase
+
+class RsUnknownCrateTypesInspectionTest : RsInspectionsTestBase(RsUnknownCrateTypesInspection::class) {
+    fun `test incorrect crate type`() = checkWarnings("""
+        #![crate_type = <error descr="Invalid `crate_type` value">"foo"</error>]
+    """)
+
+    fun `test correct crate type`() = checkWarnings("""
+        #![crate_type = "cdylib"]
+    """)
+
+    fun `test ignore if inner`() = checkWarnings("""
+        #[<error descr="identifier expected, got 'crate_type'">c</error>rate_type <error descr="'!' or '::' expected, got '='">=</error> "foo"]
```

Comment:
You need something to put the attribute on it (for example, `main` function). Otherwise, the plugins can't parse attribute correctly.
See `OuterAttr` rule in `RustParser.bnf` and its usages